exthttp/transport.go

```diff
@@ -11,6 +11,16 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+var DefaultHTTPConfig = HTTPConfig{
```

Comment:
I think this might lead to an unexpected breaking change with every other provider except S3. Do we know what the defaults for GCS currently are?

Reply:
IIUC nowadays how objstorage works then it will eventually call `defaultBaseTransport` on the google-api-go-client in [transport/http/dial.go](https://github.com/googleapis/google-api-go-client/blob/main/transport/http/dial.go#L155-L159) which actually uses `http.DefaultTransport` [src/net/http/transport.go](https://github.com/golang/go/blob/master/src/net/http/transport.go#L38-L54), they only overwrite `MaxIdleConnsPerHost` to 100 (same value we use).

So looking at the differences if we were to move forward with my change we would be setting: 

- `MaxConnsPerHost` Since this is not set on the DefaultTransport struct I believe it will just use default `int` value of `0` which would match our value
- `ResponseHeaderTimeout` Again IIUC here the default here would be `0` which would differ from our "default" of 2 minutes.
- `DialContext.DualStack`  This apparently has been [deprecated](https://github.com/golang/go/blob/master/src/net/dial.go#L98-L105) so we can even remove it from our "default" since it's enabled by default 

And we would not be setting `ForceAttemptHTTP2` which http.DefaultTransport sets. This is where we could break some things.

With all this being said I don't have much experience with all this so I'm happy to play it safe and change the PR to make GCS use `http.DefaultTransport`

Reply:
Thanks for the detailed explanation 👍 Let's try this out and see if it has any significant impact.